File: codeforces/1000-rated/HelmetsinNightLight.java
/**
someear1h 
*/
 
import java.io.*;
import java.util.*;

public class HelmetsinNightLight {

    static class Pair {
        int first;
        int second;

        Pair(int f, int s) {
            first = f;
            second = s;
        }
    }

    public static void main(String[] args) throws IOException {
        var br = new BufferedReader(new InputStreamReader(System.in));
        try(var out = new PrintWriter(System.out)) {
            int t = Integer.parseInt(br.readLine());
            while(t-- > 0) {
                var st = new StringTokenizer(br.readLine());

                int n = Integer.parseInt(st.nextToken());
                long p = Long.parseLong(st.nextToken());

                st = new StringTokenizer(br.readLine());
                long[] a = new long[n];
                for(int i = 0; i < n; i++) {
                    a[i] = Long.parseLong(st.nextToken());
                }

                st = new StringTokenizer(br.readLine());
                long[] b = new long[n];
                for(int i = 0; i < n; i++) {
                    b[i] = Long.parseLong(st.nextToken());
                }

                List<Pair> lis = new ArrayList<>();
                for(int i = 0; i < n; i++) {
                    lis.add(new Pair((int)b[i], (int)a[i]));
                }

                Collections.sort(lis, (x, y) -> x.first - y.first);

                int cnt = 1;
                long minCost = p;
                for(Pair pr : lis) {
                    if(n <= cnt) break;
                    
                    int cost = pr.first;
                    int cap = pr.second;

                    if(cost >= p) {
                        break;
                    }

                    long toTell = Math.min(cap, n-cnt);

                    minCost += toTell * cost;
                    cnt += toTell;
                }

                if(n > cnt) {
                    minCost += p * (n-cnt);
                }

                out.println(minCost);
            }
        }
    }
}

File: codeforces/1000-rated/HelmetsinNightLight.cpp
/**
* someear1h 
*/

#include <bits/stdc++.h>
using namespace std;

using ll = long long;

#define all(x) (x).begin(), (x).end()
#define pb push_back

void solve() {
    int n, p;
    cin >> n >> p;

    vector<ll> a(n), b(n);
    for(int i = 0; i < n; i++) {
        cin >> a[i];
    }

    for(int i = 0; i < n; i++) {
        cin >> b[i];
    }

    vector<pair<ll, ll>> pairs(n);
    for(int i = 0; i < n; i++) {
        pairs[i].first = b[i];
        pairs[i].second = a[i];
    }

    sort(all(pairs));

    ll shared = 1, minCost = p;

    for(auto& it : pairs) {
        ll canShare = it.second;
        ll cost = it.first;

        if(cost >= p) break;

        if(canShare + shared > n) {
            minCost += (n-shared) * cost;
            shared=n;
            break;
        } else {
            minCost += canShare * cost;
            shared += canShare;
        }
    }

    minCost += (n-shared) * p;
    cout << minCost << "\n";
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t = 1;
    cin >> t;
    while (t--) {
        solve();
    }
    return 0;
}